mcserver/views.py

```diff
@@ -570,130 +788,219 @@ def download(self, request, pk):
         url_name="async_session_download"
     )
     def async_download(self, request, pk):
-        if request.user.is_authenticated:
-            session = get_object_or_404(Session, pk=pk, user=request.user)
-            task = download_session_archive.delay(session.id, request.user.id)
-        else:
-            session = get_object_or_404(Session, pk=pk, public=True)
-            task = download_session_archive.delay(session.id)
+        try:
+            if request.user.is_authenticated:
+                session = get_object_or_404(Session, pk=pk, user=request.user)
+                task = download_session_archive.delay(session.id, request.user.id)
+            else:
+                session = get_object_or_404(Session, pk=pk, public=True)
+                task = download_session_archive.delay(session.id)
+        except Exception:
+            if settings.DEBUG:
+                raise Exception(_("error") % {"error_message": str(traceback.format_exc())})
+            raise APIException(_('session_download_error'))
+
         return Response({"task_id": task.id}, status=200)
     
     @action(detail=True)
-    def get_session_permission(self, request, pk): 
-        session = Session.objects.get(pk=pk)
-
-        isSessionOwner = session.user == request.user
-        isSessionPublic = session.public
-        isUserAdmin = request.user.groups.filter(name='admin').exists()
-        sessionPermission = {'isOwner':isSessionOwner,
-                             'isPublic':isSessionPublic,
-                             'isAdmin':isUserAdmin}
-        
-        return Response(sessionPermission)
```

Comment:
we forgot to return something from the new function

Reply:
This may have something that github (or me) changed while solving conflicts, I just added it.